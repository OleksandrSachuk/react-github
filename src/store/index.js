import { createStore, applyMiddleware } from 'redux';
import { createEpicMiddleware } from 'redux-observable';
import rootEpic from '../modules/github/epics';

import rootReducer from '../modules/github/reducers';

const epicMiddleware = createEpicMiddleware();

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  applyMiddleware(epicMiddleware),
);

epicMiddleware.run(rootEpic);

export default store;