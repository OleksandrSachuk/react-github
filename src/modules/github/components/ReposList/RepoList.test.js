import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import RepoList from './';
import store from '../../../../store';

it('renders RepoList without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RepoList store={ store }/>, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('RepoList snapshot test', () => {

  const component = renderer.create(
    <RepoList store={ store }/>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
