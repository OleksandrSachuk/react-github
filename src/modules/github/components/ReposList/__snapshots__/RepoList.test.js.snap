// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoList snapshot test 1`] = `null`;
